gcovr/configuration.py

```diff
@@ -566,6 +567,7 @@ def merge_options_and_set_defaults(partial_namespaces, all_options=None):
         help="Keep only gcov data files that match this filter. "
              "Can be specified multiple times.",
         action="append",
+        type=FilterOption,
```

Comment:
This option (and `gcov_excludes` as well) previously didn't use the `check_non_empty()` function. I wasn't sure if that was deliberate or not, but in the current iteration of the code that will be a slight behavioral change. I had to modify a test case to deal with this fact.

Reply:
I'll have to think a bit more about this.

An empty filter would be a regex that always matches, and can be used to deactivate a filter. This is most important for the `--filter` option to deactivate filtering (the default would filter to files under `--root`).

But I don't think this would be useful for `--gcov-filter` or `--gcov-exclude`: it doesn't make sense to exclude all files, and accepting all files already is the default. That only leaves use cases like “deactivating filters in a configuration file” which I'm not interested in supporting at the moment.